### DISCUSSION TITLE
How to install RPM on RHEL/CentOS/Rocky 8?

### DISCUSSION THREAD

**Discussion by tjordahl** [microsoft/openjdk#206] [Q&A] [answered]
I am interested in installing the MS OpenJDK 11 on [Rocky Linux](https://rockylinux.org/) (aka RHEL/CentOS 8).

TL;DR - I am able to hack around and get the JDK installed, but can Microsoft provide an RPM that works on RHEL/CentOS 8.x?

I tried following the CentOS 7 instructions [here](https://docs.microsoft.com/en-us/java/openjdk/install#install-on-centos-7-rpm) but the first error is:
```package packages-microsoft-prod-1.0-1.noarch does not verify: no digest```

So adding the following switches to rpm got past that:
```
sudo rpm -Uvh --nodigest --nofiledigest https://packages.microsoft.com/config/centos/7/packages-microsoft-prod.rpm
```
So far so good, but trying to install the msopenjdk-11 package:
`sudo dnf install msopenjdk-11`

results in the following:
```
Downloading Packages:
msopenjdk-11-11.0.13+8-LTS-1.x86_64.rpm          34 MB/s | 184 MB     00:05    
--------------------------------------------------------------------------------
Total                                            34 MB/s | 184 MB     00:05     
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                        1/1 
  Installing       : msopenjdk-11-11.0.13+8_LTS-1.x86_64                    1/1 
Error unpacking rpm package msopenjdk-11-11.0.13+8_LTS-1.x86_64
  Verifying        : msopenjdk-11-11.0.13+8_LTS-1.x86_64                    1/1 

Failed:
  msopenjdk-11-11.0.13+8_LTS-1.x86_64                                           

Error: Transaction failed
```
Suspecting that the Java rpm had the same problem with missing digests, I did the following:
```
sudo dnf -v --downloaddir=/tmp --downloadonly install msopenjdk-11
sudo rpm -ivh --nodigest --nofiledigest /tmp/msopenjdk-11-11.0.13+8-LTS-1.x86_64.rpm 
```
which worked:
```
$ java -version
openjdk version "11.0.13" 2021-10-19 LTS
OpenJDK Runtime Environment Microsoft-27990 (build 11.0.13+8-LTS)
OpenJDK 64-Bit Server VM Microsoft-27990 (build 11.0.13+8-LTS, mixed mode)
```

Can Microsoft provide an RPM that works on the newer 8.x versions of RHEL/CentOS (and by extension, Rocky Linux)?
Thanks
Tom

**Comment by karianna** [comment, marked as answer]
@tjordahl - Thanks for raising this and the detailed explanation!  We'll have a chat internally and get back to you.

**Reply by tjordahl** [reply]
Thanks @karianna !

**Reply by tjordahl** [reply]
@karianna Any updates?  
Not looking for a new kit right away, but perhaps a time-frame or decision (i.e. we aren't going to fix this).

**Reply by brunoborges** [reply]
@d3r3kk did the CentOS test work?

**Reply by brunoborges** [reply]
@tjordahl I just tested Rocky Linux in a container image, and I was able to install both Microsoft JDK 11 and JDK 17

**Reply by tjordahl** [reply]
@brunoborges Can you please post the exact commands you used to add the repo and `yum install` the rpm?
I could not get the latest RPM installed on Rocky 8.5 with either `rpm` or `yum install`:

```
$ sudo rpm -ivh /tmp/msopenjdk-11-11.0.13+8-LTS-1.x86_64.rpm 
Verifying...                          ################################# [100%]
Preparing...                          ################################# [100%]
Updating / installing...
   1:msopenjdk-11-11.0.13+8_LTS-1     ################################# [100%]
error: unpacking of archive failed on file /usr/lib/jvm/msopenjdk-11/bin/jaotc;6205309e: cpio: Digest mismatch
error: msopenjdk-11-11.0.13+8_LTS-1.x86_64: install failed
```

```
$ sudo rpm -e packages-microsoft-prod-1.0-1.noarch
$ sudo rpm -Uvh https://packages.microsoft.com/config/centos/8/packages-microsoft-prod.rpm
Retrieving https://packages.microsoft.com/config/centos/8/packages-microsoft-prod.rpm
Verifying...                          ################################# [100%]
Preparing...                          ################################# [100%]
Updating / installing...
   1:packages-microsoft-prod-1.0-1    ################################# [100%]
$ sudo yum install msopenjdk-11
Last metadata expiration check: 0:00:33 ago on Thu 10 Feb 2022 10:28:42 AM EST.
Dependencies resolved.
================================================================================
 Package         Arch      Version         Repository                      Size
================================================================================
Installing:
 msopenjdk-11    x86_64    11.0.11+9-1     packages-microsoft-com-prod    184 M

Transaction Summary
================================================================================
Install  1 Package

Total download size: 184 M
Installed size: 309 M
Is this ok [y/N]: y
Downloading Packages:
msopenjdk-11-11.0.11+9-1.x86_64.rpm              25 MB/s | 184 MB     00:07    
--------------------------------------------------------------------------------
Total                                            25 MB/s | 184 MB     00:07     
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                        1/1 
  Installing       : msopenjdk-11-11.0.11+9-1.x86_64                        1/1 
Error unpacking rpm package msopenjdk-11-11.0.11+9-1.x86_64
  Verifying        : msopenjdk-11-11.0.11+9-1.x86_64                        1/1 

Failed:
  msopenjdk-11-11.0.11+9-1.x86_64                                               

Error: Transaction failed
```

**Reply by brunoborges** [reply]
In my case it works because I used `centos/7/` for the packages.microsoft.com URL.

At the end of the day, you can use `centos/7/` because MS Build of OpenJDK RPM has no dependency on any other Microsoft-specific package.

The MS JDK packages on `centos/8/` should not be there. It is an old build of JDK 11 (11.0.11). I'll take note to delete it.

The main reason we are not publishing to _CentOS 8_ is that this version is no longer supported - and it was never an LTS release either. The future is **CentOS Stream**.

```bash
bruno@MSPC-BRBORGES:~$ docker run -ti --rm rockylinux
[root@d109a3475d2b /]# rpm -Uvh https://packages.microsoft.com/config/centos/8/packages-microsoft-prod.rpm
Retrieving https://packages.microsoft.com/config/centos/7/packages-microsoft-prod.rpm
Verifying...                          ################################# [100%]
Preparing...                          ################################# [100%]
Updating / installing...
   1:packages-microsoft-prod-1.0-1    ################################# [100%]
[root@d109a3475d2b /]# yum install msopenjdk-17
Rocky Linux 8 - AppStream                                             4.4 MB/s | 9.1 MB     00:02
Rocky Linux 8 - BaseOS                                                4.2 MB/s | 5.6 MB     00:01
Rocky Linux 8 - Extras                                                 15 kB/s |  12 kB     00:00
packages-microsoft-com-prod                                           5.6 MB/s | 4.2 MB     00:00
Dependencies resolved.
======================================================================================================
 Package                      Arch        Version              Repository                        Size
======================================================================================================
Installing:
 msopenjdk-17                 x86_64      17+35-1              packages-microsoft-com-prod      182 M
Installing dependencies:
 alsa-lib                     x86_64      1.2.5-4.el8          appstream                        488 k
 dejavu-fonts-common          noarch      2.35-7.el8           baseos                            73 k
 dejavu-sans-fonts            noarch      2.35-7.el8           baseos                           1.5 M
 fontconfig                   x86_64      2.13.1-4.el8         baseos                           273 k
 fontpackages-filesystem      noarch      1.44-22.el8          baseos                            15 k
 freetype                     x86_64      2.9.1-4.el8_3.1      baseos                           393 k
 libX11                       x86_64      1.6.8-5.el8          appstream                        610 k
 libX11-common                noarch      1.6.8-5.el8          appstream                        157 k
 libXau                       x86_64      1.0.9-3.el8          appstream                         36 k
 libXext                      x86_64      1.3.4-1.el8          appstream                         44 k
 libXi                        x86_64      1.7.10-1.el8         appstream                         48 k
 libXrender                   x86_64      0.9.10-7.el8         appstream                         32 k
 libXtst                      x86_64      1.2.3-7.el8          appstream                         21 k
 libpng                       x86_64      2:1.6.34-5.el8       baseos                           125 k
 libxcb                       x86_64      1.13.1-1.el8         appstream                        228 k

Transaction Summary
======================================================================================================
Install  16 Packages

Total download size: 186 M
Installed size: 328 M
Is this ok [y/N]: y
Downloading Packages:
(1/16): libX11-1.6.8-5.el8.x86_64.rpm                                 668 kB/s | 610 kB     00:00
(2/16): libX11-common-1.6.8-5.el8.noarch.rpm                          168 kB/s | 157 kB     00:00
(3/16): libXau-1.0.9-3.el8.x86_64.rpm                                 341 kB/s |  36 kB     00:00
(4/16): libXext-1.3.4-1.el8.x86_64.rpm                                404 kB/s |  44 kB     00:00
(5/16): alsa-lib-1.2.5-4.el8.x86_64.rpm                               446 kB/s | 488 kB     00:01
(6/16): libXi-1.7.10-1.el8.x86_64.rpm                                 497 kB/s |  48 kB     00:00
(7/16): libXrender-0.9.10-7.el8.x86_64.rpm                            283 kB/s |  32 kB     00:00
(8/16): libXtst-1.2.3-7.el8.x86_64.rpm                                151 kB/s |  21 kB     00:00
(9/16): dejavu-fonts-common-2.35-7.el8.noarch.rpm                     698 kB/s |  73 kB     00:00
(10/16): libxcb-1.13.1-1.el8.x86_64.rpm                               1.5 MB/s | 228 kB     00:00
(11/16): fontpackages-filesystem-1.44-22.el8.noarch.rpm               106 kB/s |  15 kB     00:00
(12/16): fontconfig-2.13.1-4.el8.x86_64.rpm                           1.1 MB/s | 273 kB     00:00
(13/16): dejavu-sans-fonts-2.35-7.el8.noarch.rpm                      1.1 MB/s | 1.5 MB     00:01
(14/16): libpng-1.6.34-5.el8.x86_64.rpm                               107 kB/s | 125 kB     00:01
(15/16): freetype-2.9.1-4.el8_3.1.x86_64.rpm                          280 kB/s | 393 kB     00:01
(16/16): msopenjdk-17-17+35-1.x86_64.rpm                               12 MB/s | 182 MB     00:15
------------------------------------------------------------------------------------------------------
Total                                                                  10 MB/s | 186 MB     00:18
Rocky Linux 8 - AppStream                                             1.6 MB/s | 1.6 kB     00:00
Importing GPG key 0x6D745A60:
 Userid     : "Release Engineering <infrastructure@rockylinux.org>"
 Fingerprint: 7051 C470 A929 F454 CEBE 37B7 15AF 5DAC 6D74 5A60
 From       : /etc/pki/rpm-gpg/RPM-GPG-KEY-rockyofficial
Is this ok [y/N]: y
Key imported successfully
packages-microsoft-com-prod                                           4.3 kB/s | 983  B     00:00
Importing GPG key 0xBE1229CF:
 Userid     : "Microsoft (Release signing) <gpgsecurity@microsoft.com>"
 Fingerprint: BC52 8686 B50D 79E3 39D3 721C EB3E 94AD BE12 29CF
 From       : https://packages.microsoft.com/keys/microsoft.asc
Is this ok [y/N]: y
Key imported successfully
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                              1/1
  Installing       : fontpackages-filesystem-1.44-22.el8.noarch                                  1/16
  Installing       : dejavu-fonts-common-2.35-7.el8.noarch                                       2/16
  Installing       : dejavu-sans-fonts-2.35-7.el8.noarch                                         3/16
  Installing       : libpng-2:1.6.34-5.el8.x86_64                                                4/16
  Installing       : freetype-2.9.1-4.el8_3.1.x86_64                                             5/16
  Installing       : fontconfig-2.13.1-4.el8.x86_64                                              6/16
  Running scriptlet: fontconfig-2.13.1-4.el8.x86_64                                              6/16
  Installing       : libXau-1.0.9-3.el8.x86_64                                                   7/16
  Installing       : libxcb-1.13.1-1.el8.x86_64                                                  8/16
  Installing       : libX11-common-1.6.8-5.el8.noarch                                            9/16
  Installing       : libX11-1.6.8-5.el8.x86_64                                                  10/16
  Installing       : libXext-1.3.4-1.el8.x86_64                                                 11/16
  Installing       : libXi-1.7.10-1.el8.x86_64                                                  12/16
  Installing       : libXtst-1.2.3-7.el8.x86_64                                                 13/16
  Installing       : libXrender-0.9.10-7.el8.x86_64                                             14/16
  Installing       : alsa-lib-1.2.5-4.el8.x86_64                                                15/16
  Running scriptlet: alsa-lib-1.2.5-4.el8.x86_64                                                15/16
  Installing       : msopenjdk-17-17+35-1.x86_64                                                16/16
  Running scriptlet: msopenjdk-17-17+35-1.x86_64                                                16/16
  Running scriptlet: fontconfig-2.13.1-4.el8.x86_64                                             16/16
  Verifying        : alsa-lib-1.2.5-4.el8.x86_64                                                 1/16
  Verifying        : libX11-1.6.8-5.el8.x86_64                                                   2/16
  Verifying        : libX11-common-1.6.8-5.el8.noarch                                            3/16
  Verifying        : libXau-1.0.9-3.el8.x86_64                                                   4/16
  Verifying        : libXext-1.3.4-1.el8.x86_64                                                  5/16
  Verifying        : libXi-1.7.10-1.el8.x86_64                                                   6/16
  Verifying        : libXrender-0.9.10-7.el8.x86_64                                              7/16
  Verifying        : libXtst-1.2.3-7.el8.x86_64                                                  8/16
  Verifying        : libxcb-1.13.1-1.el8.x86_64                                                  9/16
  Verifying        : dejavu-fonts-common-2.35-7.el8.noarch                                      10/16
  Verifying        : dejavu-sans-fonts-2.35-7.el8.noarch                                        11/16
  Verifying        : fontconfig-2.13.1-4.el8.x86_64                                             12/16
  Verifying        : fontpackages-filesystem-1.44-22.el8.noarch                                 13/16
  Verifying        : freetype-2.9.1-4.el8_3.1.x86_64                                            14/16
  Verifying        : libpng-2:1.6.34-5.el8.x86_64                                               15/16
  Verifying        : msopenjdk-17-17+35-1.x86_64                                                16/16

Installed:
  alsa-lib-1.2.5-4.el8.x86_64                          dejavu-fonts-common-2.35-7.el8.noarch
  dejavu-sans-fonts-2.35-7.el8.noarch                  fontconfig-2.13.1-4.el8.x86_64
  fontpackages-filesystem-1.44-22.el8.noarch           freetype-2.9.1-4.el8_3.1.x86_64
  libX11-1.6.8-5.el8.x86_64                            libX11-common-1.6.8-5.el8.noarch
  libXau-1.0.9-3.el8.x86_64                            libXext-1.3.4-1.el8.x86_64
  libXi-1.7.10-1.el8.x86_64                            libXrender-0.9.10-7.el8.x86_64
  libXtst-1.2.3-7.el8.x86_64                           libpng-2:1.6.34-5.el8.x86_64
  libxcb-1.13.1-1.el8.x86_64                           msopenjdk-17-17+35-1.x86_64

Complete!
[root@d109a3475d2b /]# java -version
openjdk version "17" 2021-09-14
OpenJDK Runtime Environment Microsoft-26987 (build 17+35)
OpenJDK 64-Bit Server VM Microsoft-26987 (build 17+35, mixed mode, sharing)
[root@d109a3475d2b /]#
```

And this is for installing JDK 11:

```bash
bruno@MSPC-BRBORGES:~$ docker run -ti --rm rockylinux
[root@2ed0719d5aa6 /]# rpm -Uvh https://packages.microsoft.com/config/centos/7/packages-microsoft-prod.rpm
Retrieving https://packages.microsoft.com/config/centos/7/packages-microsoft-prod.rpm
Verifying...                          ################################# [100%]
Preparing...                          ################################# [100%]
Updating / installing...
   1:packages-microsoft-prod-1.0-1    ################################# [100%]
[root@2ed0719d5aa6 /]# yum install msopenjdk-11
Rocky Linux 8 - AppStream                                             4.8 MB/s | 9.1 MB     00:01
Rocky Linux 8 - BaseOS                                                1.4 MB/s | 5.6 MB     00:03
Rocky Linux 8 - Extras                                                 14 kB/s |  12 kB     00:00
packages-microsoft-com-prod                                           4.9 MB/s | 4.2 MB     00:00
Dependencies resolved.
======================================================================================================
 Package                      Arch        Version              Repository                        Size
======================================================================================================
Installing:
 msopenjdk-11                 x86_64      11.0.14+9_LTS-1      packages-microsoft-com-prod      184 M
Installing dependencies:
 alsa-lib                     x86_64      1.2.5-4.el8          appstream                        488 k
 dejavu-fonts-common          noarch      2.35-7.el8           baseos                            73 k
 dejavu-sans-fonts            noarch      2.35-7.el8           baseos                           1.5 M
 fontconfig                   x86_64      2.13.1-4.el8         baseos                           273 k
 fontpackages-filesystem      noarch      1.44-22.el8          baseos                            15 k
 freetype                     x86_64      2.9.1-4.el8_3.1      baseos                           393 k
 libX11                       x86_64      1.6.8-5.el8          appstream                        610 k
 libX11-common                noarch      1.6.8-5.el8          appstream                        157 k
 libXau                       x86_64      1.0.9-3.el8          appstream                         36 k
 libXext                      x86_64      1.3.4-1.el8          appstream                         44 k
 libXi                        x86_64      1.7.10-1.el8         appstream                         48 k
 libXrender                   x86_64      0.9.10-7.el8         appstream                         32 k
 libXtst                      x86_64      1.2.3-7.el8          appstream                         21 k
 libpng                       x86_64      2:1.6.34-5.el8       baseos                           125 k
 libxcb                       x86_64      1.13.1-1.el8         appstream                        228 k

Transaction Summary
======================================================================================================
Install  16 Packages

Total download size: 188 M
Installed size: 322 M
Is this ok [y/N]: y
Downloading Packages:
(1/16): libX11-common-1.6.8-5.el8.noarch.rpm                          253 kB/s | 157 kB     00:00
(2/16): alsa-lib-1.2.5-4.el8.x86_64.rpm                               758 kB/s | 488 kB     00:00
(3/16): libX11-1.6.8-5.el8.x86_64.rpm                                 893 kB/s | 610 kB     00:00
(4/16): libXau-1.0.9-3.el8.x86_64.rpm                                 354 kB/s |  36 kB     00:00
(5/16): libXext-1.3.4-1.el8.x86_64.rpm                                529 kB/s |  44 kB     00:00
(6/16): libXi-1.7.10-1.el8.x86_64.rpm                                 636 kB/s |  48 kB     00:00
(7/16): libXtst-1.2.3-7.el8.x86_64.rpm                                264 kB/s |  21 kB     00:00
(8/16): libXrender-0.9.10-7.el8.x86_64.rpm                            347 kB/s |  32 kB     00:00
(9/16): libxcb-1.13.1-1.el8.x86_64.rpm                                2.3 MB/s | 228 kB     00:00
(10/16): dejavu-fonts-common-2.35-7.el8.noarch.rpm                    840 kB/s |  73 kB     00:00
(11/16): fontconfig-2.13.1-4.el8.x86_64.rpm                           2.8 MB/s | 273 kB     00:00
(12/16): fontpackages-filesystem-1.44-22.el8.noarch.rpm               192 kB/s |  15 kB     00:00
(13/16): libpng-1.6.34-5.el8.x86_64.rpm                               1.4 MB/s | 125 kB     00:00
(14/16): freetype-2.9.1-4.el8_3.1.x86_64.rpm                          3.3 MB/s | 393 kB     00:00
(15/16): dejavu-sans-fonts-2.35-7.el8.noarch.rpm                      4.0 MB/s | 1.5 MB     00:00
(16/16): msopenjdk-11-11.0.14+9-LTS-1.x86_64.rpm                       19 MB/s | 184 MB     00:09
------------------------------------------------------------------------------------------------------
Total                                                                  17 MB/s | 188 MB     00:10
Rocky Linux 8 - AppStream                                             1.6 MB/s | 1.6 kB     00:00
Importing GPG key 0x6D745A60:
 Userid     : "Release Engineering <infrastructure@rockylinux.org>"
 Fingerprint: 7051 C470 A929 F454 CEBE 37B7 15AF 5DAC 6D74 5A60
 From       : /etc/pki/rpm-gpg/RPM-GPG-KEY-rockyofficial
Is this ok [y/N]: y
Key imported successfully
packages-microsoft-com-prod                                           4.0 kB/s | 983  B     00:00
Importing GPG key 0xBE1229CF:
 Userid     : "Microsoft (Release signing) <gpgsecurity@microsoft.com>"
 Fingerprint: BC52 8686 B50D 79E3 39D3 721C EB3E 94AD BE12 29CF
 From       : https://packages.microsoft.com/keys/microsoft.asc
Is this ok [y/N]: y
Key imported successfully
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                              1/1
  Installing       : fontpackages-filesystem-1.44-22.el8.noarch                                  1/16
  Installing       : dejavu-fonts-common-2.35-7.el8.noarch                                       2/16
  Installing       : dejavu-sans-fonts-2.35-7.el8.noarch                                         3/16
  Installing       : libpng-2:1.6.34-5.el8.x86_64                                                4/16
  Installing       : freetype-2.9.1-4.el8_3.1.x86_64                                             5/16
  Installing       : fontconfig-2.13.1-4.el8.x86_64                                              6/16
  Running scriptlet: fontconfig-2.13.1-4.el8.x86_64                                              6/16
  Installing       : libXau-1.0.9-3.el8.x86_64                                                   7/16
  Installing       : libxcb-1.13.1-1.el8.x86_64                                                  8/16
  Installing       : libX11-common-1.6.8-5.el8.noarch                                            9/16
  Installing       : libX11-1.6.8-5.el8.x86_64                                                  10/16
  Installing       : libXext-1.3.4-1.el8.x86_64                                                 11/16
  Installing       : libXi-1.7.10-1.el8.x86_64                                                  12/16
  Installing       : libXtst-1.2.3-7.el8.x86_64                                                 13/16
  Installing       : libXrender-0.9.10-7.el8.x86_64                                             14/16
  Installing       : alsa-lib-1.2.5-4.el8.x86_64                                                15/16
  Running scriptlet: alsa-lib-1.2.5-4.el8.x86_64                                                15/16
  Installing       : msopenjdk-11-11.0.14+9_LTS-1.x86_64                                        16/16
  Running scriptlet: msopenjdk-11-11.0.14+9_LTS-1.x86_64                                        16/16
  Running scriptlet: fontconfig-2.13.1-4.el8.x86_64                                             16/16
  Verifying        : alsa-lib-1.2.5-4.el8.x86_64                                                 1/16
  Verifying        : libX11-1.6.8-5.el8.x86_64                                                   2/16
  Verifying        : libX11-common-1.6.8-5.el8.noarch                                            3/16
  Verifying        : libXau-1.0.9-3.el8.x86_64                                                   4/16
  Verifying        : libXext-1.3.4-1.el8.x86_64                                                  5/16
  Verifying        : libXi-1.7.10-1.el8.x86_64                                                   6/16
  Verifying        : libXrender-0.9.10-7.el8.x86_64                                              7/16
  Verifying        : libXtst-1.2.3-7.el8.x86_64                                                  8/16
  Verifying        : libxcb-1.13.1-1.el8.x86_64                                                  9/16
  Verifying        : dejavu-fonts-common-2.35-7.el8.noarch                                      10/16
  Verifying        : dejavu-sans-fonts-2.35-7.el8.noarch                                        11/16
  Verifying        : fontconfig-2.13.1-4.el8.x86_64                                             12/16
  Verifying        : fontpackages-filesystem-1.44-22.el8.noarch                                 13/16
  Verifying        : freetype-2.9.1-4.el8_3.1.x86_64                                            14/16
  Verifying        : libpng-2:1.6.34-5.el8.x86_64                                               15/16
  Verifying        : msopenjdk-11-11.0.14+9_LTS-1.x86_64                                        16/16

Installed:
  alsa-lib-1.2.5-4.el8.x86_64                          dejavu-fonts-common-2.35-7.el8.noarch
  dejavu-sans-fonts-2.35-7.el8.noarch                  fontconfig-2.13.1-4.el8.x86_64
  fontpackages-filesystem-1.44-22.el8.noarch           freetype-2.9.1-4.el8_3.1.x86_64
  libX11-1.6.8-5.el8.x86_64                            libX11-common-1.6.8-5.el8.noarch
  libXau-1.0.9-3.el8.x86_64                            libXext-1.3.4-1.el8.x86_64
  libXi-1.7.10-1.el8.x86_64                            libXrender-0.9.10-7.el8.x86_64
  libXtst-1.2.3-7.el8.x86_64                           libpng-2:1.6.34-5.el8.x86_64
  libxcb-1.13.1-1.el8.x86_64                           msopenjdk-11-11.0.14+9_LTS-1.x86_64

Complete!
[root@2ed0719d5aa6 /]# java -version
openjdk version "11.0.14" 2022-01-18 LTS
OpenJDK Runtime Environment Microsoft-30257 (build 11.0.14+9-LTS)
OpenJDK 64-Bit Server VM Microsoft-30257 (build 11.0.14+9-LTS, mixed mode)
[root@2ed0719d5aa6 /]#
```

**Reply by brunoborges** [reply]
Alternatively, you can simply download the `tar.gz` package.

**Reply by brunoborges** [reply]
Regarding support, this is our policy: https://docs.microsoft.com/en-us/java/openjdk/support#operating-systems-and-linux-distributions

**Reply by tjordahl** [reply]
I seem to have found the problem.  I am running my Rocky in FIPS mode (`fips-mode-setup --enable`).  This mode seems to require a different digest algorithm in the RPM. I assume one that is FIPS validated.  If I turn off FIPS mode, the Microsoft RPMs install without issue.

See https://access.redhat.com/discussions/5043121

> It seems like, once you enable FIPS mode on RHEL 8, RPM-validation becomes fairly hardcore. If a vendor supplies an RPM that is unsigned or weakly-signed or has digests that use use weak algorithms, yum/dnf will refuse to install the package.

The [workaround referenced in the thread](https://access.redhat.com/discussions/6333871) for the Oracle RPM is to create a macro to avoid the file digest:
```
echo "%_pkgverify_level none" >/etc/rpm/macros.verify
```
This does not work for me with the MS package however, as I still get an error unpacking.  This seems to be something that is common with CentOS/RHEL 7 built packages.  The Elasticsearch team encountered and fixed the issue - see https://github.com/elastic/elasticsearch/issues/58257 .  I see this FIPS/RHEL8 problem being addressed in other packages out there as well ([example here](https://gitlab.com/gitlab-org/omnibus-gitlab/-/issues/5938)).

I hope this information helps the team resolve this issue.  As I stated before, this isn't an urgent need for me right now as Rocky/RHEL 8.x is in the future for me, but when we do move there, we will need an RPM that can be installed in FIPS mode.

Thanks for your help @brunoborges !

**Reply by brunoborges** [reply]
@tjordahl we will continue looking into this through issue https://github.com/microsoft/openjdk/issues/222
